spring-cloud-cloudfoundry-discovery/src/main/java/org/springframework/cloud/cloudfoundry/discovery/reactive/CloudFoundryNativeReactiveDiscoveryClient.java

```diff
@@ -85,15 +86,25 @@ protected ServiceInstance mapApplicationInstanceToServiceInstance(
 		String instanceId = applicationId + "." + applicationIndex;
 		String name = applicationDetail.getName();
 		String url = applicationDetail.getUrls().size() > 0
-				? applicationDetail.getUrls().get(0) : null;
-		boolean secure = (url + "").toLowerCase().startsWith("https");
```

Comment:
Why did you change this?

Reply:
- getUrls().get(0) always returns the external URL. It's not picking up the apps.internal url when we use CloudFoundryAppServiceReactiveDiscoveryClient
- protocol information is never returned as part of the client call. URL info always comes with the protocol.